Windows: Menhir dependency fails to build

**esy version:** `master` + Windows fixes
**Operating System:** Windows

**Issue:** 
`menhir` fails to install with this error when running the `ocamlbuild -where` command:
```
     'C:\Users\bryph\.esy\3_\b\opam__slash__menhir-20171013.0.0-05503f0d\src\_stage1'
         The system cannot find the path specified.
         Warning: Won't be able to compile a native plugin
         Failure: Cannot find "ocamlbuild.cmo" in ocamlbuild -where
     directory.
         make[2]: *** [Makefile:38: stage1] Error 2
```
Some notes on my environment:
 - __`ocamlbuild`__ seems to be installed successfully - it is available here:  `C:/Users/bryph/.esy/3_/i/opam__slash__ocamlbuild-0.12.0-5e55ac68/bin`

Running __`ocamlbuild -where` from the build environment gives some strange results - it returns `C:/Users/bryph/.esy/3_/s/opam__slash__ocamlbuild-0.12.0-5e55ac68/lib/ocamlbuild`, which is the _staging_ folder. I validated that the library `ocamlbuild.cmo` is _actually available_ in the correct folder, the `install` folder: `C:/Users/bryph/.esy/3_/i/opam__slash__ocamlbuild-0.12.0-5e55ac68/lib/ocamlbuild`

So the mystery that needs to be solved is - why is `ocamlbuild -where` pointing to the _staging_ folder instead of the _install_ folder? It's possible that something isn't wired up quite right in the environment.

One other thing I noticed, that may or may not be related, is that the `esy-installer` step failed for `ocamlbuild`:
```
# esy-build-package: running: "sh" "-c" "(esy-installer || true)"

[DEBUG]: Running command: "node" "E:/esy-bash/bin/esy-bash.js" "--env" "E:/test-file9_1531265326..dat" "sh" "-c" "(esy-installer || true)"

esy-bash: executing bash command: sh "-c" "(esy-installer || true)" | nonce: 1
 -- using environment file: E:/test-file9_1531265326..dat
 -- script file: C:/Users/bryph/AppData/Local/Temp/__esy-bash__1531265326858__1__.sh
ocamlbuild.byte                  => C:/Users/bryph/.esy/3_/s/opam__slash__ocamlbuild-0.12.0-5e55ac68/bin/ocamlbuild.byte
[ERROR] Failure during install
# opam-version         1.2.2
# os                   cygwin
# command              install -m 0755 /cygdrive/c/Users/bryph/.esy/3_/b/opam__slash__ocamlbuild-0.12.0-5e55ac68/ocamlbuild.byte C:/Users/bryph/.esy/3_/s/opam__slash__ocamlbuild-0.12.0-5e55ac68/bin/ocamlbuild.byte
# path                 /cygdrive/c/Users/bryph/.esy/3_/b/opam__slash__ocamlbuild-0.12.0-5e55ac68
# exit-code            1
# env-file             /tmp/opam-5568/log/log-5568-5a61c5.env
# stdout-file          /tmp/opam-5568/log/log-5568-5a61c5.out
# stderr-file          /tmp/opam-5568/log/log-5568-5a61c5.err
### stderr ###
# install: cannot create regular file 'C:/Users/bryph/.esy/3_/s/opam__slash__ocamlbuild-0.12.0-5e55ac68/bin/ocamlbuild.byte': File exists
esy-bash: process exited with code 0
````


__Minimal `package.json` repro:__ 
```
{
  "name": "test-menhir",
  "version": "0.0.0",
  "esy": {
    "build": [
      "true"
    ]
  },
  "dependencies": {
    "@opam/menhir": "*"
  },
  "devDependencies": {
    "ocaml": "esy-ocaml/ocaml#6aacc05"
  }
}
```

--- a/esy-build-package/Build.re
+++ b/esy-build-package/Build.re
@@ -335,7 +335,7 @@ let commitBuildToStore = (config: Config.t, build: build) => {
         let%bind () = rewritePrefixInFile(~origPrefix=origPrefixString, ~destPrefix=destPrefixString, path);
         let normalizedOrigPrefix = Path.normalizePathSlashes(origPrefixString);
         let normalizedDestPrefix = Path.normalizePathSlashes(destPrefixString);
-        let%bind () = rewritePrefixInFile(~origPrefix=normalizedOrigPrefixPath, ~destPrefix=normalizedDestPrefixPath, path);
+        let%bind () = rewritePrefixInFile(~origPrefix=normalizedOrigPrefix, ~destPrefix=normalizedDestPrefix, path);
         ok;
     | _ =>
         rewritePrefixInFile(~origPrefix=origPrefixString, ~destPrefix=destPrefixString, path);

--- a/esy-build-package/Build.re
+++ b/esy-build-package/Build.re
@@ -320,25 +320,25 @@ let commitBuildToStore = (config: Config.t, build: build) => {
         empty
         % config.fastreplacestringCmd
         % p(path)
-        % p(origPrefix)
-        % p(destPrefix)
+        % origPrefix
+        % destPrefix
       );
     Bos.OS.Cmd.run(cmd);
   };
   let rewritePrefixesInFile = (~origPrefix, ~destPrefix, path) => {
+    let origPrefixString = Path.to_string(origPrefix);
+    let destPrefixString = Path.to_string(destPrefix);
     switch (System.Platform.host) {
     | Windows =>
         /* On Windows, the slashes could be either `/` or windows-style `\` */
         /* We'll replace both styles */
-        let%bind () = rewritePrefixInFile(~origPrefix, ~destPrefix, path);
-        let normalizedOrigPrefix = Path.normalizePathSlashes(Path.to_string(origPrefix));
-        let normalizedDestPrefix = Path.normalizePathSlashes(Path.to_string(destPrefix));
-        let normalizedOrigPrefixPath = Fpath.v(normalizedOrigPrefix);
-        let normalizedDestPrefixPath = Fpath.v(normalizedDestPrefix);
+        let%bind () = rewritePrefixInFile(~origPrefix=origPrefixString, ~destPrefix=destPrefixString, path);
+        let normalizedOrigPrefix = Path.normalizePathSlashes(origPrefixString);
+        let normalizedDestPrefix = Path.normalizePathSlashes(destPrefixString);
         let%bind () = rewritePrefixInFile(~origPrefix=normalizedOrigPrefixPath, ~destPrefix=normalizedDestPrefixPath, path);
         ok;
     | _ =>
-        rewritePrefixInFile(~origPrefix, ~destPrefix, path);
+        rewritePrefixInFile(~origPrefix=origPrefixString, ~destPrefix=destPrefixString, path);
     }
   };
   let rewriteTargetInSymlink = (~origPrefix, ~destPrefix, path) => {

--- a/esy-build-package/Build.re
+++ b/esy-build-package/Build.re
@@ -325,6 +325,22 @@ let commitBuildToStore = (config: Config.t, build: build) => {
       );
     Bos.OS.Cmd.run(cmd);
   };
+  let rewritePrefixesInFile = (~origPrefix, ~destPrefix, path) => {
+    switch (System.Platform.host) {
+    | Windows =>
+        /* On Windows, the slashes could be either `/` or windows-style `\` */
+        /* We'll replace both styles */
+        let%bind () = rewritePrefixInFile(~origPrefix, ~destPrefix, path);
+        let normalizedOrigPrefix = Path.normalizePathSlashes(Path.to_string(origPrefix));
+        let normalizedDestPrefix = Path.normalizePathSlashes(Path.to_string(destPrefix));
+        let normalizedOrigPrefixPath = Fpath.v(normalizedOrigPrefix);
+        let normalizedDestPrefixPath = Fpath.v(normalizedDestPrefix);
+        let%bind () = rewritePrefixInFile(~origPrefix=normalizedOrigPrefixPath, ~destPrefix=normalizedDestPrefixPath, path);
+        ok;
+    | _ =>
+        rewritePrefixInFile(~origPrefix, ~destPrefix, path);
+    }
+  };
   let rewriteTargetInSymlink = (~origPrefix, ~destPrefix, path) => {
     let%bind targetPath = readlink(path);
     switch (Path.rem_prefix(origPrefix, targetPath)) {
@@ -339,7 +355,7 @@ let commitBuildToStore = (config: Config.t, build: build) => {
   let relocate = (path: Path.t, stats: Unix.stats) =>
     switch (stats.st_kind) {
     | Unix.S_REG =>
-      rewritePrefixInFile(
+      rewritePrefixesInFile(
         ~origPrefix=build.stagePath,
         ~destPrefix=build.installPath,
         path,

